File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/answer/Answer.java
package com.github.dedis.popstellar.model.network.answer;

import com.github.dedis.popstellar.model.network.GenericMessage;

import java.util.Objects;

/**
 * An abstract result from a request
 *
 * <p>Is linked to an earlier request with a unique id
 */
public abstract class Answer extends GenericMessage {

  private final int id;

  /**
   * Constructor of an Answer
   *
   * @param id of the answer
   */
  public Answer(int id) {
    this.id = id;
  }

  /** Returns the ID of the answer */
  public int getId() {
    return id;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Answer answer = (Answer) o;
    return getId() == answer.getId();
  }

  @Override
  public int hashCode() {
    return Objects.hash(getId());
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/answer/Answer.kt
package com.github.dedis.popstellar.model.network.answer

import com.github.dedis.popstellar.model.network.GenericMessage
import java.util.Objects

/**
 * An abstract result from a request
 *
 * Is linked to an earlier request with a unique id
 */
abstract class Answer
/**
 * Constructor of an Answer
 *
 * @param id of the answer
 */
(
    /** Returns the ID of the answer */
    val id: Int
) : GenericMessage() {

  override fun equals(other: Any?): Boolean {
    if (this === other) {
      return true
    }
    if (other == null || javaClass != other.javaClass) {
      return false
    }
    val answer = other as Answer
    return id == answer.id
  }

  override fun hashCode(): Int {
    return Objects.hash(id)
  }
}
